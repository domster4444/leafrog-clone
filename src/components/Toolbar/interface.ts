export interface IProps {
  logo: string;
  links: {
    name: string;
    link: string;
  }[];
}
