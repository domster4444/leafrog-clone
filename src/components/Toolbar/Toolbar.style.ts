import styled from 'styled-components';

interface IProps {
  color?: string;
}

export const Nav = styled.nav<IProps>`
  background: var(--primary-white);
  display: flex;
  padding: 1.5rem 0rem;
  justify-content: space-between;
`;

export const NavList = styled.ul`
  list-style: none;
  display: flex;
  align-items: center;
`;
export const NavItem = styled.li`
  margin: 0.85rem;
  &:hover {
    a {
      color: var(--primary-navy);
    }
  }
`;
export const NavLink = styled.a`
  text-decoration: none;
  color: var(--primary-black);
  font-size: 1.8rem;
`;

export const Logo = styled.img`
  height: 4rem;
  width: 20rem;
`;
