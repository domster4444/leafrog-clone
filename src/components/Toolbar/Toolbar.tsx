import React from 'react';
import { IProps } from './interface';
import { Nav, NavList, NavLink, NavItem, Logo } from './Toolbar.style';

/**
 * @param  components
 */
import Button from 'components/Button';

const Toolbar: React.FC<IProps> = ({ logo, links }): React.ReactElement => {
  return (
    <>
      <header>
        <Nav data-testid="toolbar">
          <NavList>
            {logo && (
              <NavItem>
                <Logo src={logo} alt="logo" />
              </NavItem>
            )}
          </NavList>
          <NavList>
            {links.map((item, index) => {
              return (
                <NavItem key={index}>
                  <NavLink href="#">{item.name} </NavLink>
                </NavItem>
              );
            })}
          </NavList>
          <NavList>
            <NavItem>
              <Button type="tertiary">GET STARTED</Button>
            </NavItem>
          </NavList>
        </Nav>
      </header>
    </>
  );
};

export default Toolbar;
