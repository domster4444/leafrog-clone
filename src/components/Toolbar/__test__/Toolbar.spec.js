import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
/**
 * @imports components
 */
import Toolbar from 'components/Toolbar';

const componentProps = {
  logo: 'https://www.lftechnology.com/wp-content/themes/Froggy/img/logo_leapfrog.svg',
  links: [
    { name: 'What We Do', link: '#' },
    { name: 'Services', link: '#' },
    { name: 'Insights', link: '#' },
    { name: 'Resources', link: '#' },
    { name: 'About Us', link: '#' },
  ],
};

const renderComponent = (logo, links) => {
  render(<Toolbar logo={logo} links={links} />);
};

describe('Toolbar Component', () => {
  test('check if toolbar is rendered', () => {
    renderComponent(componentProps.logo, componentProps.links);
    expect(screen.getByTestId('toolbar')).toBeInTheDocument();
  });

  test('check if it has a logo ', () => {
    renderComponent(componentProps.logo, componentProps.links);
    expect(screen.getByRole('img')).toBeInTheDocument();
  });
  test('check if it has menus', () => {
    renderComponent(componentProps.logo, componentProps.links);
    expect(screen.getAllByRole('link')).not.toBe(null);
  });
  test('check if it has button', () => {
    renderComponent(componentProps.logo, componentProps.links);
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  test('should have a button with the correct text', () => {
    renderComponent(componentProps.logo, componentProps.links);
    expect(screen.getByRole('button')).toHaveTextContent(/GET STARTED/i);
  });

  test('this component should not change', () => {
    const tree = TestRenderer.create(
      <Toolbar logo={componentProps.logo} links={componentProps.links} />
    );

    expect(tree).toMatchSnapshot();
  });
});
