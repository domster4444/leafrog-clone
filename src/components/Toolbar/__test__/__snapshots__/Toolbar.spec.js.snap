// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Component this component should not change 1`] = `
<header>
  <nav
    className="sc-bczRLJ jYBnCD"
    data-testid="toolbar"
  >
    <ul
      className="sc-gsnTZi fTNjvb"
    >
      <li
        className="sc-dkzDqf eqExpN"
      >
        <img
          alt="logo"
          className="sc-eCYdqJ jZHEog"
          src="https://www.lftechnology.com/wp-content/themes/Froggy/img/logo_leapfrog.svg"
        />
      </li>
    </ul>
    <ul
      className="sc-gsnTZi fTNjvb"
    >
      <li
        className="sc-dkzDqf eqExpN"
      >
        <a
          className="sc-hKMtZM eqRkyw"
          href="#"
        >
          What We Do
           
        </a>
      </li>
      <li
        className="sc-dkzDqf eqExpN"
      >
        <a
          className="sc-hKMtZM eqRkyw"
          href="#"
        >
          Services
           
        </a>
      </li>
      <li
        className="sc-dkzDqf eqExpN"
      >
        <a
          className="sc-hKMtZM eqRkyw"
          href="#"
        >
          Insights
           
        </a>
      </li>
      <li
        className="sc-dkzDqf eqExpN"
      >
        <a
          className="sc-hKMtZM eqRkyw"
          href="#"
        >
          Resources
           
        </a>
      </li>
      <li
        className="sc-dkzDqf eqExpN"
      >
        <a
          className="sc-hKMtZM eqRkyw"
          href="#"
        >
          About Us
           
        </a>
      </li>
    </ul>
    <ul
      className="sc-gsnTZi fTNjvb"
    >
      <li
        className="sc-dkzDqf eqExpN"
      >
        <button
          className="sc-jSMfEi fsrtFw"
        >
          GET STARTED
        </button>
      </li>
    </ul>
  </nav>
</header>
`;
